src/gc/impl/conservative/gc.d

```diff
@@ -395,6 +398,8 @@ class ConservativeGC : GC
             alloc_size = size;
         }
         gcx.leakDetector.log_malloc(p, size);
+        bytesAllocated += alloc_size;
```

Comment:
Thread locals can be slow. Versioning this under a compile time condition might be acceptable.

Reply:
Compile time is not a good option IMHO, because then you can't use this feature with a normal compiler. Two solutions come in mind. One is doing the proxy trick and providing some sort of flag to pick the GC you want to use, the tracing GC being one of the options. I don't remember if that mechanism is already in place and if currently any other GC is provided by the official distribution. Another option would be to do a simple runtime `if (tracing) bytesAllocated += alloc_size;`, that should be fast enough for all users.

Reply:
The cost is there but should be evaluated relative to the surrounding time. I recall I measured a while ago and an allocation took several thousand cycles even in the best case. The additional overhead would be difficult to measure.

Reply:
I did a short measurement with the vdparser benchmark, adding the TLS counter costs about 1% for Win64 (which probably has one of the faster implementation of TLS), though that measurement was done on a mobile processor with disabled speedstepping. Thats probably also consistent with the full allocation taking about 1000 cycles, given the additional instructions generated.